### DISCUSSION TITLE
[REQ] "AI" audio tools to integrate & more

### DISCUSSION THREAD

**Discussion by MarcoRavich** [intel/openvino-plugins-ai-audacity#77] [Ideas]
Hi there, thanks for this **very** interesting set of free audio plugins !

It would be extremely useful to have more "AI" functions - such as multitrack synching, automixing, guitar sim, DRC, etc. - available (and not only for audacity).

We're collecting some in our [HyMPS project](https://forart.it/HyMPS/) (under the [AUDIO section](https://github.com/forart/HyMPS#-) \ [AI-based page](https://github.com/forart/HyMPS/blob/main/Audio/AI-based.md#--)), but here's some so-called "awesome" resources list to carefully check:

- https://github.com/ybayle/awesome-deep-learning-music
- https://github.com/skirdey/awesome-deep-audio
- https://github.com/archinetai/audio-ai-timeline
- https://github.com/EmulationAI/awesome-large-audio-models

Last but not least, it would be great to have a more "generic" version (exploiting [JUCE](https://github.com/juce-framework/JUCE) maybe ?) of this plugins pack to allow its use in users' favorite editing softwares.

Hope that inspires !

**Comment by RyanMetcalfeInt8** [comment]
Hi @forart,

Thank you for your interest in this project! And also, thanks for posting all of those great resources -- I've already started to go through some of the lists -- they are quite comprehensive.

I'll take a look at JUCE -- it would certainly be advantageous to somehow encapsulate these plugins such that they are compatible with more frameworks. The trouble that I typically have with more 'generic' plugin frameworks is that somehow limit some of the capabilities, but certainly open to take another look.

Best Regards!
Ryan

**Comment by MarcoRavich** [comment]
> Hi @forart,
> I'll take a look at JUCE -- it would certainly be advantageous to somehow encapsulate these plugins such that they are compatible with more frameworks.

Well, @leomccormack's [SPARTA](https://leomccormack.github.io/sparta-site/) is a good example to have an idea of what is possible to "manage" with JUCE:

![](https://raw.githubusercontent.com/leomccormack/SPARTA/master/sparta_screenshot.png)

Hope that inspires.

**Comment by MarcoRavich** [comment]
Just mentioned this discussion in [Audacity's Real-time-capable limiter issue](https://github.com/audacity/audacity/issues/6305#issuecomment-2074441949): what about "porting" @jamesnapierstuart's [AI Compressor](https://github.com/jamesnapierstuart/19329-AI-Compressor#19329-ai-compressor) within OV plugins?

note: I've also created a specific [Effects subsection](https://github.com/forart/HyMPS/blob/main/Audio/AI-based.md#effects-) in [HyMPS](https://forart.it/HyMPS/)'s [AI-based page](https://github.com/forart/HyMPS/blob/main/Audio/AI-based.md#--), where this project is listed too.

**Comment by MarcoRavich** [comment]
Bump.

Just found an interesting open project that may help/inspire: @hazenluk's [Real-time Modeling of Nonlinear Dynamical Systems for Analog Audio Emulation](https://github.com/hazenluk/ArtificialAnalog), a _Neural Network Based Modelling of Audio Effects, **Using JUCE**_.

Its Youtube Demo shows the model being used in a VST environment:
[![Diode Clipper Distortion VST Example](https://i3.ytimg.com/vi/AVldGj0ZkN4/original.jpg)](https://youtu.be/AVldGj0ZkN4 "Watch Diode Clipper Distortion VST Example")

**Comment by MarcoRavich** [comment]
Bump.

Well, the most popular and discussed AI-based audio tool these days is undoubtedly @sdatkinson's [Neural Amp Modeler](https://github.com/sdatkinson/neural-amp-modeler#readme) / [Plug-in](https://github.com/sdatkinson/NeuralAmpModelerPlugin) (VST3/AU built with [iPlug2](https://iplug2.github.io/)) even if it uses G-Colab service for training.

It would be interesting for Intel-based PC owners to have an offline OpenVINO-based version too.

Hope that inspires.

**Comment by MarcoRavich** [comment]
Bump.

This might be the "columbus egg": 
@yeeking's [ai-enhanced-audio-book](https://github.com/yeeking/ai-enhanced-audio-book#readme) - A repository of AI-enhanced audio plugins written using C++, JUCE, libtorch, RTNeural and other libraries
